### PR TITLE
Sort targets by instance id for deterministic ordering

### PR DESCRIPTION
**Pull request by xichen2020**
cc @cw9 @prateek @jeromefroe 

This PR sorts the deployment targets in each deployment step by their instance IDs for deterministic ordering so it's easier to see in logs.

### CODE REVIEW

File: tools/deploy/planner.go [hunk 1]
```diff
@@ -135,6 +136,8 @@ func (p deploymentPlanner) generateStepFromTargetType(
 			delete(instances, shardSetID)
 			continue
 		}
+
+		done := false
```
Comment (jeromefroe) [inline tools/deploy/planner.go]: nit: use default initialization, ie `var done bool`

Comment (xichen2020) [reply tools/deploy/planner.go]: Yeah I tend to be more explicit in cases like this, feels more clear to the readers.

Comment (cw9) [reply tools/deploy/planner.go]: nit: it feels cleaner to create a generateSortedStepFromTargetType that sorts the step generated from generateStepFromTargetType

Comment (xichen2020) [reply tools/deploy/planner.go]: Hm that feels like a bit overkill, this is only sorting instances within the same step so whether they sorted or not doesn't affect correctness of the output, so might as well always sort them for more deterministic output.

Comment (cw9) [reply tools/deploy/planner.go]: sounds good

**Comment by cw9** [comment]
LGTM

**Pull request merged**